autonat.go

```diff
@@ -336,6 +374,23 @@ func (as *AmbientAutoNAT) probeNextPeer() {
 		if proto, err := as.host.Peerstore().SupportsProtocols(p, AutoNATProto); len(proto) == 0 || err != nil {
 			continue
 		}
+
+		goodAddr := false
+		for _, a := range info.Addrs {
+			if !as.config.skipDial(a) {
+				goodAddr = true
+				// if a public IP of the peer is one of ours: skip the peer.
+				aIP, _ := manet.ToIP(a)
```

Comment:
This is messy. We're checking this once inside `skipDial`, then trying to re-check to figure out why we returned `false`. We should either:

1. Pass the entire `AddrInfo` into `skipDial` and let it figure it out (simpler?).
2. Have `skipDial` return an enum (yes, no, continue).